torch/csrc/jit/python/pybind_utils.h

```diff
@@ -144,6 +144,36 @@ struct VISIBILITY_HIDDEN PythonFutureWrapper
         PyObjectType::get()));
   }
 
+  void add_done_callback(py::function cb) {
+    auto pf = std::make_shared<PythonFunctionGuard>(std::move(cb));
+    fut->addCallback(std::bind(
+        [pyFut(this->getPtr())](std::shared_ptr<PythonFunctionGuard> pf) {
+          try {
+            pybind11::gil_scoped_acquire ag;
+            pf->func_(pyFut);
+          } catch (py::error_already_set& e) {
+            {
+              pybind11::gil_scoped_acquire ag;
+              // Release ownership on py::objects and also restore Python
+              // Error Indicator.
+              e.restore();
+              // Clear the Python Error Indicator as we has recorded the
+              // exception in the response message.
+              PyErr_Clear();
+            }
+            // Log and ignore exceptions raised through the callback
+            std::cout << "Got the following error when running the callback: "
```

Comment:
usually, we don't do `cout`. If you would like to log, can we use sth like this:

https://github.com/pytorch/pytorch/blob/8cd3857bc7c57bd412941812f40fdb3a9dc6dd45/torch/csrc/distributed/rpc/tensorpipe_agent.cpp#L250